src/server/storage/storageAPI.js

```diff
@@ -38,16 +41,45 @@ const setup = (app: Router, storage: StorageAPI) => {
       )
         throw new Error('User email or mobile not verified!')
 
-      const user: UserRecord = defaults(body.user, {
+      const user: UserRecord = defaults(bodyUser, {
         identifier: userRecord.loggedInAs,
         createdDate: new Date().toString()
       })
+
       //mautic contact should already exists since it is first created during the email verification we update it here
       const mauticRecord = process.env.NODE_ENV === 'development' ? {} : await Mautic.createContact(user).catch(e => {})
       const mauticId = get(mauticRecord, 'contact.fields.all.id', -1)
       logger.debug('User mautic record', { mauticId, mauticRecord })
+
+      const updateUserObj = {
+        ...user,
+        mauticId
+      }
+
+      if (!user.w3Token) {
```

Comment:
this is going to be false also for users that did came from W3.
if the api on W3 will work ok also for existing emails that's fine... will it?

Reply:
You mean that here is needed to check if such email already exists on web3?